packages/solid-query/src/createBaseQuery.ts

```diff
@@ -76,7 +68,21 @@ export function createBaseQuery<
     })
   })
 
-  onCleanup(() => unsubscribe())
+  createComputed(() => {
+    batch(() => {
+      if (state.fetchStatus === 'idle') {
+        res()
+        res = noop
+      } else {
+        refetch()
+      }
+    })
```

Comment:
```suggestion
    if (state.fetchStatus === 'idle') {
      res()
      res = noop
    } else {
      refetch()
    }
```

We don't need a batch here